frontend/src/slices/apiSlice.ts

```diff
@@ -25,7 +23,9 @@ export const apiSlice = createApi({
     "Messages",
     "MqttMessages",
   ],
-  baseQuery: fetchBaseQuery({ baseUrl: `${apiBaseURI}/api/v1` }),
+  baseQuery: fetchBaseQuery({
+    baseUrl: `${window.location.origin}/api/v1`,
```

Comment:
This doesn't work if you are running the SPA and the API separately but not gated through nginx/caddy. Shouldn't this receive the correct base uri?

Reply:
The Vite dev server will proxy this correctly since it's not actually nested under `/next`.

Reply:
Right, but run the production frontend on port 8000 and the api on port 9000, and do not front it with nginx or caddy. And you'll see what I'm talking about.

Reply:
yeah agree there, can we push this off for now until we don't have to deal with subpath issues? Else I think we can just use env vars to pass to the FE build: https://vitejs.dev/guide/env-and-mode

Reply:
@kevinelliott done! Also added CORS support to the uvicorn server